index.src.html

```diff
@@ -279,59 +285,91 @@
 
+    {{Credential}}'s default implementation of {{Credential/[[CollectFromCredentialStore]](origin, options)}}:
+
     <ul class="algorithm">
       1.  Return an empty set.
     </ul>
-  </section>
 
-  <section algorithm="'discover credentials' internal method">
+
+  <h5 id="algorithm-discover-creds" algorithm>`[[DiscoverFromExternalSource]]` internal method</h5>
     <dfn for="Credential" method export>\[[DiscoverFromExternalSource]](origin, options)</dfn> is called
-    with an [=environment settings object/origin=] and a {{CredentialRequestOptions}} object. 
+    with an [=environment settings object/origin=] and a {{CredentialRequestOptions}} object.
     It returns a {{Credential}} if one can be
     returned given the options provided, `null` if no credential is available, or an error if
     discovery fails (for example, incorrect options could produce a {{TypeError}}). If this
     kind of {{Credential}} is only [=effective=] for a single use or a limited time, this
     method is responsible for generating new [=credentials=] using a [=credential source=].
 
+    {{Credential}}'s default implementation of {{Credential/[[DiscoverFromExternalSource]](origin, options)}}:
+
     <ul class="algorithm">
       1.  Return `null`.
     </ul>
-  </section>
-  
-  <section algorithm="'store a credential' internal method">
+
+  <h5 id="algorithm-store-cred" algorithm>`[[Store]]` internal method</h5>
     <dfn for="Credential" method export>\[[Store]](credential)</dfn> is called with a {{Credential}},
-    and returns once {{Credential}} is persisted to the [=credential store=]:
+    and returns once {{Credential}} is persisted to the [=credential store=].
+
+    {{Credential}}'s default implementation of {{Credential/[[Store]](credential)}}:
 
     <ul class="algorithm">
       1.  Return.
     </ul>
-  </section>
 
-  <section algorithm="'create a credential' internal method">
-    <dfn for="Credential" method export>\[[Create]](options)</dfn> is called with a
-    {{CredentialCreationOptions}}, and returns either a {{Credential}}, if one can be created, 
-    `null` if no credential was created, or an error if creation fails due to exceptional situations
-    (for example, incorrect options could produce a {{TypeError}}):
+  <h5 id="algorithm-create-cred" algorithm>`[[Create]]` internal method</h5>
+    <dfn for="Credential" method export>\[[Create]](origin, options)</dfn> is called 
+    [=in parallel=] with an [=environment settings object/origin=] and a {{CredentialCreationOptions}},
+    and either:
+
+    * creates a {{Credential}}, or, 
+    * does not create a credential and returns `null`, or,
+    * returns an error if creation fails due to exceptional situations
+        (for example, incorrect options could produce a {{TypeError}}).
+
+    When creating a {{Credential}}, it will either:
+
+    * directly return a {{Credential}} (in which case it accomplished everything while
```

Comment:
I think this is impossible to do safely: creating instances of interface types needs to connect them to a Realm via a JS global, and I think we can't do that from *in parallel*. How about having all successful [[Create]] calls return the algorithm instead of {{Credential}} instances?

Reply:
hm, oh yeah huh. @mikewest ?

Reply:
The simplest change might be to have {Password,Federated}Credential.[[Create]] "return an algorithm that does the following steps" instead of doing the steps themselves. Then in a subsequent change, there might be some pieces that should move back to the *in parallel* section.

Reply:
I'm fine with altering the algorithms to return algorithms to return data.

That said, it's a little frustrating that that's necessary, and it would be nice if we could define some reasonable way of saying "Do the thing you, dear reader, know that I mean when I write 'return this thing'." It's not clear that we're adding much explanatory value to the spec by jumping through these hoops.